context/context.go

```diff
@@ -0,0 +1,67 @@
+package context
+
+import (
+	gocontext "context"
+	"fmt"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Context interface {
+	Errorf(err error)
+	StartSpan(tracer trace.Tracer, spanName string) trace.Span
+	SetSpanAttributes(key string, val any)
+}
```

Comment:
A lot of the commonly used methods on the context cannot be added here. It's only these 3 that can be shared

- to avoid kubernetes dependency
```
Namespace() string
Kubernetes() kubernetes.Interface
```
- All `With*` methods cannot be added here as it returns the common's context.

```	
WithDB(db *gorm.DB) Context
WithEchoContext(ctx EchoContext) Context
WithContext(ctx gocontext.Context) Context
WithTimeout(timeout time.Duration) (Context, func())
WithUser(user *ContextUser) Context
User() *ContextUser
```

- requires duty import

```
GetEnvVarValue(input types.EnvVar) (string, error)
GetEnvValueFromCache(env types.EnvVar) (string, error)
HydrateConnection(connectionIdentifier string) (*models.Connection, error)
```